plugins/module_utils/ec2.py

```diff
@@ -455,7 +455,8 @@ def boto3_tag_list_to_ansible_dict(tags_list, tag_name_key_name=None, tag_value_
     else:
         tag_candidates = {'key': 'value', 'Key': 'Value'}
 
-    if not tags_list:
+    # minio seems to return [{}] as an empty tags_list
+    if not tags_list or not tags_list[0]:
```

Comment:
I'd make sure it's entirely empty, since otherwise there should be an error below
```suggestion
    if not tags_list or not any(tag for tag in tags_list):
```

Reply:
At this point we could just 'repair' taglist:
```suggestion
    tags_list = [tag for tag in tags_list if tag]
    if not tags_list:
```
This should even provide any further code to choke on '{}'.

Reply:
I was thinking the error was helpful for developers so they could see they were passing in a bad list.

Reply:
My thinking was being permissive helps to build a more resilient tool (There is not much to misinterpret about an empty dict here, right?).
But i'm fine with either solution. Which one should i pick?

Reply:
@jillr @tremble Which one do you prefer?

Reply:
I'm good with either option, However, I would defer to @s-hertel's experience.

We recently added some minimal [unit tests for boto3_tag_list_to_ansible_dict](https://github.com/ansible-collections/amazon.aws/blob/main/tests/unit/module_utils/test_ec2.py#L393).  To reduce the chance of this being reverted by someone unfamiliar with minio would you be willing to add some tests for the 'empty' behaviour?

Should be something like:
```
    def test_boto3_tag_list_to_ansible_dict_empty(self):
        # AWS returns [] when there are no tags
        self.assertEqual(boto3_tag_list_to_ansible_dict([], {})
        # Minio returns [{}] when there are no tags
        self.assertEqual(boto3_tag_list_to_ansible_dict([{}], {})
```

Reply:
@tremble thanks for designing the test for me.

Reply:
As someone who's only been contributing to Ansible for a year or so I'm all too aware of how daunting it can be to be asked to write the initial tests for a module or some of the shared code when all you wanted to do was add a minor feature/fix.  I've been trying to improve the situation but it's slow going (especially when writing tests uncovers bugs)